crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -252,6 +273,13 @@ impl ReadView {
         direction: IterDirection,
     ) -> impl Stream<Item = StorageResult<Message>> + '_ {
         futures::stream::iter(self.on_chain.all_messages(start_message_id, direction))
+            .chunks(self.batch_size)
+            .filter_map(|chunk| async move {
+                // Give a chance to other tasks to run.
+                tokio::task::yield_now().await;
```

Comment:
Does the stream get treated like one giant future instead of a serious of separate futures? Then I'd understand why we need this, but seems kinda hacky for a situation that seems pretty innocuous.